### PR TITLE
fix: Validate embedding output dimensionality.

### PR DESCRIPTION
**Pull request by lee1premium**
## Description

Fixes: b/349273965

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [X] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by holtskinner — APPROVED** [review]

File: ai-platform/snippets/predict-text-embeddings.js [hunk 1]
```diff
@@ -44,11 +44,12 @@ async function main(
     const [response] = await client.predict(request);
     console.log('Got predict response');
     const predictions = response.predictions;
-    for (const prediction of predictions) {
-      const embeddings = prediction.structValue.fields.embeddings;
-      const values = embeddings.structValue.fields.values.listValue.values;
-      console.log('Got prediction: ' + JSON.stringify(values));
-    }
+    const embeddings = predictions.map(p =>
+      p.structValue.fields.embeddings.structValue.fields.values.listValue.values.map(
```
Comment (skarukas) [inline ai-platform/snippets/predict-text-embeddings.js:48]: Can this be split up to make the line and statement shorter?
```js
p => {
  const embeddingsProto = p.structValue.fields.embeddings
  const valuesProto = structValue.fields.values
  return valuesProto.listValue.values.map(v => v.numberValue)
}
```

(Not sure if they're protos or some other value representation)

File: ai-platform/snippets/predict-text-embeddings-preview.js [hunk 2]
```diff
@@ -42,13 +42,13 @@ async function main(
...
     const [response] = await client.predict(request);
-    console.log('Got predict response');
     const predictions = response.predictions;
-    for (const prediction of predictions) {
-      const embeddings = prediction.structValue.fields.embeddings;
-      const values = embeddings.structValue.fields.values.listValue.values;
-      console.log('Got prediction: ' + JSON.stringify(values));
-    }
+    const embeddings = predictions.map(p =>
+      p.structValue.fields.embeddings.structValue.fields.values.listValue.values.map(
```
Comment (skarukas) [inline ai-platform/snippets/predict-text-embeddings-preview.js:47]: Same here.

**Review by skarukas — APPROVED** [review]

**Review by yan283 — APPROVED** [review]

File: ai-platform/snippets/predict-text-embeddings-preview.js (hunk 2 above)
Comment (lee1premium) [reply ai-platform/snippets/predict-text-embeddings-preview.js]: Done.

File: ai-platform/snippets/predict-text-embeddings.js (hunk 1 above)
Comment (lee1premium) [reply ai-platform/snippets/predict-text-embeddings.js]: They are embeddingsStructProto and valuesListProto.

**Pull request merged**